### DISCUSSION TITLE
How to handle FirebaseAuth signIn after signOut? Getting permission-denied in Firestore.

### DISCUSSION THREAD

**Discussion by melanki** [firebase/flutterfire#7304] [Q&A]
Trying to achieve a simple signOut and signIn in with Firesbase Auth, but after relogging Firestore gives permission denied.

1. AuthService.signOut()
2. Navigator.push => Login screen
3. AuthService.signInWithEmailAndPassword
4. Navigator.push => Home
5. Navigator.push => DataScreen (onPressed)
6. Error

"[cloud_firestore/permission-denied]The caller does not have permission to execute the specified operation."

This seems to be due to Firestore not getting the updated state of new user that logs in, and thus giving permission-denied as it is trying to access the collection without any authentication?

**Comment by codeOfJannik** [comment]
I have the same issue. After logout and relogging I always get the permission denied error for my snapshot.listen() calls. Did you already find any solution @melanki?

**Reply by melanki** [reply]
I am using Riverpod for state management and solved this by refreshing the firestore provider,
which is basically the same as creating a new firestore instance every time you switch firebase auth user.

**Reply by codeOfJannik** [reply]
Just for your information. I was able to resolve the issue. Updating to the most recent firebase package versions did the job for me :)
